src/Network/Benchmark.hs

```diff
@@ -139,19 +138,20 @@ runAction
         , MonadBaseControl IO m
         , MonadIO m
         )
-    ⇒ Stat
+    ⇒ Int -- ^ number of retries
+    → Stat
     → TestAction
     → m Stat
-runAction stat action = timeT run >>= \case
-    (t, Right ()) →
-        return $ seq stat $ stat ⊕ successStat (toSeconds t * 1000)
-    (t, Left e) →
-        return $ seq stat $ stat ⊕ failStat (toSeconds t * 1000) (sshow e)
+runAction retries stat action = do
+    (t, result) ← timeT $ runEitherT run
+    return ∘ seq stat ∘ (stat ⊕) $
+        case result of
+            Right () → successStat $ toSeconds t * 1000
+            Left e → failStat (toSeconds t * 1000) (sshow e)
```

Comment:
Fair enough... I tend to prefer duplicated stuff factored out, but I see your point :)

I think perhaps why the old one might have been more readable is that the intuitive order of events was better represented. I'm happy to change it back if you want, since the adjustment was mostly mechanical.